projects/igniteui-angular/src/lib/navigation-drawer/themes/_base.scss

```diff
@@ -0,0 +1,265 @@
+    }
+
+    @include b(igx-nav-drawer) {
+        --igx-nav-drawer-size: #{rem(240px)};
+
+        flex-basis: 0;
+        transition: flex-basis;
+        transition-duration: .3s;
+        transition-timing-function: $ease-out-quad;
+        flex-shrink: 0;
+
+        @include m(pinned) {
+            flex-basis: var(--igx-nav-drawer-size);
+        }
+
+        @include mx(mini, pinned) {
+            flex-basis: calc(var(--igx-mini-nav-drawer-size, #{var(--igx-nav-drawer-size--mini)}) + rem(1px));
+        }
+
+        @include e(aside) {
+            position: fixed;
+            height: 100%;
+            min-height: 100%;
+            overflow-x: hidden;
+            background: var-get($_theme, 'background');
+            top: 0;
+            bottom: 0;
+            width: var(--igx-nav-drawer-size);
+            inset-inline-start: 0;
+            z-index: 999;
+            transition: width, padding, transform;
+            transition-timing-function: $ease-in-out-quad;
+            box-shadow: var-get($_theme, 'elevation');
+            padding: var(--aside-padding);
+            border-inline-end: rem(1px) solid var-get($_theme, 'border-color');
+            border-radius: var-get($_theme, 'border-radius');
+
+            &.panning {
+                overflow-x: hidden;
+                transition: none;
+            }
+        }
+
+        @include e(aside, collapsed) {
+            transform: translate3d(-300px, 0, 0);
+
+            [dir='rtl'] & {
+                transform: translate3d(300px, 0, 0);
+            }
+
+            box-shadow: none;
+
+            &.igx-nav-drawer__aside--right {
+                transform: translate3d(300px, 0, 0);
+                box-shadow: none;
+
+                [dir='rtl'] & {
+                    transform: translate3d(-300px, 0, 0);
+                }
+            }
+        }
+
+        @include e(aside, right) {
+            inset-inline-start: auto;
+            inset-inline-end: 0;
+            border-inline-end: none;
+            border-inline-start: rem(1px) solid var-get($_theme, 'border-color');
+        }
+
+        @include e(aside, mini) {
+            transition-duration: .3s;
+            width: #{var(--igx-nav-drawer-size--mini)};
+            min-width: fit-content;
+
+            %item {
+                justify-content: center;
+                min-width: fit-content;
+
+                igx-icon {
+                    margin-inline-start: 0;
+                }
+            }
+        }
+
+        @include e(aside, normal) {
+            transition-duration: .3s;
+            width: var(--igx-nav-drawer-size);
+        }
+
+        @include e(aside, pinned) {
+            position: relative;
+            box-shadow: none;
+            z-index: 0;
+
+            &.igx-nav-drawer__aside--collapsed {
+                transform: none;
+                width: 0;
+                overflow: hidden;
+                border: none;
+                padding: 0;
+            }
```

Comment:
If you want to apply some style rules when the aside is both pinned and collapsed you should use the $mods syntax:

```scss
@include e(aside, $mods: (pinned, collapsed) {
/* ... style rules */
}
```